scripts/image-output-extractor/qiskit_docs_notebook_image_extractor/__init__.py

```diff
@@ -0,0 +1,135 @@
+# This code is a Qiskit project.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+import nbformat
+import base64
+import shutil
+from subprocess import Popen, PIPE
+from pathlib import Path
+from dataclasses import dataclass
+from itertools import chain
+
+NOTEBOOK_PATHS = chain(
+    Path("docs").rglob("*.ipynb"),
+    Path("learning").rglob("*.ipynb"),
+)
+
+
+@dataclass
+class SvgImage:
+    data: str
+    filepath: Path
+
+    def write(self):
+        self.filepath.write_text(self.data)
+
+
+@dataclass
+class RasterImage:
+    data: bytes
+    filepath: Path
+
+    def write(self):
+        self.filepath.write_bytes(self.data)
+
+
+Image = SvgImage | RasterImage
+
+
+def main():
+    """
+    Search for notebooks and extract image outputs if necessary.
+    """
+    for nb_path in NOTEBOOK_PATHS:
+        images_folder = prepare_image_folder(nb_path)
+        nb = nbformat.read(nb_path, 4)
+        print(f"{nb_path}")
```

Comment:
Leftover from debugging?

Reply:
I thought it'd be good to see which files have been processed, however happy to remove. It will be hidden from `./fix` users anyway.

Reply:
I lean towards deleting it because we run on every file, even if the file was already processed. I think that's noisy given how many notebooks we have.

If you do keep it, I'd probably recommend adding a prefix like `Processed {notebook}`